Feat(spv 1451) filter transactions by dates and status

cain4chain opened the pull request:
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

cain4chain committed:
feat(SPV-1506) Update styling on mobile devices

feat(_admin) : add navigation for mobile device
feat(_user) : add navigation for mobile device
chore(_user) : add navigation link for xPub
chore(login) : update login responsiveness

cain4chain committed:
feat(SPV-1506) : update user routes

chore(_user) : Update user routes with Paymail route

cain4chain committed:
fix(styling) : search bar, xPub link

fix(styling) : fix styling issue with search bar outline hidden on top of boarder
fix(user.xpub) : remove dupicate user xpub on desktop version

cain4chain committed:
feat(transactions) : update transaction filtering

feat(_admin.transactions) : Update admin transaction page with new filtering logic

feat(_user.transactions) : Update user transaction page with new filtering logic

chore(transactionQueryOptions) : edit transaction query options to include new params for status

chore(transaction_components) : Update transaction components with predefined types for status, create constants for transaction types

cain4chain committed:
chore(lint formatting) : update formatting

chore(lint) : update linting issues
chore(prettier) : update prettier formatting

cain4chain committed:
chore(Transaction Query) : Update type variable

chore(Transaction Query) : Update transaction variable to use constant type as default

dzolt-4chain commented on src/components/TransactionsColumns/columns.tsx:
```diff
@@ -59,11 +60,21 @@ export const columns: ColumnDef<Tx>[] = [
-        <Badge variant="outline">Recorded</Badge>
-      );
+      const status = row.getValue('status') as TransactionStatusValue;
+      switch (status) {
+        case TRANSACTION_STATUS.MINED:
+          return <Badge variant="secondary">Mined</Badge>;
+        case TRANSACTION_STATUS.CREATED:
+          return <Badge variant="outline">Created</Badge>;
+        case TRANSACTION_STATUS.BROADCASTED:
+          return <Badge>Broadcasted</Badge>;
+        case TRANSACTION_STATUS.REVERTED:
+          return <Badge variant="destructive">Reverted</Badge>;
+        case TRANSACTION_STATUS.PROBLEMATIC:
+          return <Badge variant="destructive">Problematic</Badge>;
+        default:
+          return <Badge variant="outline">{status}</Badge>;
+      }
```
This is the same as in TransactionsColumnsMobile, maybe we could make a function somewhere and reuse it here since statuses will probably be the same in mobile and web version? WDYT?

cain4chain replied:
Yes good point. I will add this into a Utility function for this.

dzolt-4chain commented on src/interfaces/transaction.ts:
```diff
@@ -1,5 +1,12 @@
+import { TransactionStatusValue } from '@/constants';
+
+declare module '@bsv/spv-wallet-js-client' {
+  interface Tx {
```
Why do we need to extend the module here like that? I'm generally curious :D Doesn't js client have the statuses returning?

cain4chain replied:
It seems that the JS client contains the old values for transaction status 
https://github.com/bitcoin-sv/spv-wallet-js-client/blob/bc0dc715805b4c43210c8dd4a2bccf7e04f413d0/src/filters.ts#L52

Plus it is open to any string value as per the comment here
https://github.com/bitcoin-sv/spv-wallet-js-client/blob/bc0dc715805b4c43210c8dd4a2bccf7e04f413d0/src/filters.ts#L78

We may need to update the JS client first with the new values, and discuss whether we would like to include any String value still in the response.

dzolt-4chain replied:
All good then, as agreed task will be created to update the types in js client and change it as a part of the task

cain4chain committed:
feat(SPV-1451) : refactor changes

feat(transactionStatusBadge) :  create utility function for status badge value
eat(transactionStatusLabel) :  create utility function for status label value

cain4chain committed:
Merge branch 'main' into feat(SPV-1451)-filter-transactions-by-dates-and-status

cain4chain committed:
feat(SPV-1451) : Rebase resolution

feat(SPV-1451) : Rebase resolution for admin and user transactions. Use drop down instead of tablist components.
